### PR TITLE
Create mapping and coupling from ZDS to ZGW

### PR DESCRIPTION
**Pull request by rjzondervan**
- CreeerZaakID results in a new ZGW case with only an identifier
- CreeerZaak results in the ZGW case being populated
- CreeerDocumentId results in a new ZGW document (EnkelvoudigInformatieObject)
- CreeerDocument results in the EnkelvoudigInformatieObject being populated and a ZaakInformatieObject being made to connect the document to the case

### CODE REVIEW

File: Service/InstallationService.php [hunk 1]
```diff
@@ -28,10 +24,14 @@ class InstallationService implements InstallerInterface
     ];
 
     public const ENDPOINTS = [
-        ['path' => 'stuf/zds', 'throws' => ['vrijbrp.zds.inbound'], 'name' => 'zds-endpoint']
+        ['path' => 'stuf/zds', 'throws' => ['vrijbrp.zds.inbound'], 'name' => 'zds-endpoint', 'methods' => []],
     ];
 
     public const ACTION_HANDLERS = [
```
Comment (WilcoLouwerse) [inline Service/InstallationService.php:30]: Still not a fan of this way of creating Actions...
We can actually use action.json files now

Comment (WilcoLouwerse) [reply Service/InstallationService.php:30]: https://github.com/CommonGateway/GeboorteVrijBRPBundle/blob/a36ee0e43e3dc9842feac28247913733714a1a38/Action/vrijbrp.ZgwToVrijbrpAction.action.json

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "minimum-stability": "dev",
     "require": {
         "php": ">=7.4",
-        "commongateway/corebundle": "dev-master",
+        "commongateway/corebundle": "dev-master || dev-feature/im-24/add-json-to-array-cast",
```
Comment (WilcoLouwerse) [inline composer.json]: No

File: Service/ZdsToZgwService.php [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+     * @param array $data   The inbound data for the case
+     * @param array $config The configuration for the action
+     *
+     * @return array
+     */
+    public function documentActionHandler(array $data, array $config): array
+    {
+        $this->configuration = $config;
+
+        $zaakEntity = $this->getEntity('https://vng.opencatalogi.nl/schemas/zrc.zaak.schema.json');
```
Comment (WilcoLouwerse) [inline Service/ZdsToZgwService.php:339]: This works fine,
Just feels like getting entities or other objects by reference could/should be also be in configuration, example:
https://github.com/CommonGateway/GeboorteVrijBRPBundle/blob/a36ee0e43e3dc9842feac28247913733714a1a38/Action/vrijbrp.ZgwToVrijbrpAction.action.json

https://github.com/CommonGateway/GeboorteVrijBRPBundle/blob/a36ee0e43e3dc9842feac28247913733714a1a38/Service/ZgwToVrijbrpService.php#L162

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]
composer.json

File: Service/InstallationService.php (hunk 1 above)
Comment (rjzondervan) [reply Service/InstallationService.php:30]: This is an endpoint ;), agree on actions, but speed etc.

File: composer.json (hunk 2 above)
Comment (rjzondervan) [reply composer.json]: Will remove

**Review by WilcoLouwerse — APPROVED** [review]

File: Service/InstallationService.php (hunk 1 above)
Comment (WilcoLouwerse) [reply Service/InstallationService.php:30]: My comment is on line 34 | 30
public const ACTION_HANDLERS = [

And i get that, this is just a comment not a request to change it right now, approved!

**Review by bbrands02 — APPROVED** [review]

**Pull request merged**